qCC/ccCommandLineCommands.cpp

```diff
@@ -1348,19 +1349,114 @@ bool CommandSubsample::process(ccCommandLineInterface& cmd)
 		{
 			return cmd.error(QObject::tr("Missing parameter: spatial step after \"-%1 SPATIAL\"").arg(COMMAND_SUBSAMPLE));
 		}
+
 		bool ok;
 		double step = cmd.arguments().takeFirst().toDouble(&ok);
 		if (!ok || step <= 0)
 		{
-			return cmd.error(QObject::tr("Invalid step value for spatial resampling!"));
+			return cmd.error(QObject::tr("Invalid step value for spatial subsampling!"));
 		}
 		cmd.print(QObject::tr("\tSpatial step: %1").arg(step));
-		
+
+		double sfMinSpacing = 0;
+		double sfMaxSpacing = 0;
+		bool useActiveSF = false;
+		if (!cmd.arguments().empty())
+		{
+			if (cmd.arguments().front().toUpper() == OPTION_USE_ACTIVE_SF)
+			{
+				//enable USE_ACTIVE_SF
+				useActiveSF = true;
+				cmd.arguments().pop_front();
+				if (cmd.arguments().size() >= 2)
+				{
+					bool validMin;
+					sfMinSpacing = cmd.arguments().takeFirst().toDouble(&validMin);
+					bool validMax;
+					sfMaxSpacing = cmd.arguments().takeFirst().toDouble(&validMax);
+					if (!(validMin&&validMax) || sfMinSpacing < 0 || sfMaxSpacing < 0)
+					{
+						return cmd.error(QObject::tr("Invalid parameters: Two positive decimal number required after '%1'").arg(OPTION_USE_ACTIVE_SF));
+					}
+				}
+				else
+				{
+					return cmd.error(QObject::tr("Missing parameters: Two positive decimal number required after '%1'").arg(OPTION_USE_ACTIVE_SF));
+				}
+			}
+		}
+
 		for (CLCloudDesc& desc : cmd.clouds())
 		{
 			cmd.print(QObject::tr("\tProcessing cloud %1").arg(!desc.pc->getName().isEmpty() ? desc.pc->getName() : "no name"));
-			
+
 			CCCoreLib::CloudSamplingTools::SFModulationParams modParams(false);
+
+			//handle Use Active SF on each cloud
+			if (useActiveSF)
+			{
+				//look for the min and max sf values
+				ccScalarField* sf = desc.pc->getCurrentDisplayedScalarField();
+				if (!sf)
+				{
+					//warn the user, not use active SF and keep going
+					cmd.warning(QObject::tr("\tUse active SF error: not found active scalar field. Set one with '-%1'").arg(COMMAND_SET_ACTIVE_SF));
```

Comment:
More generally, I would treat this kind of error as real 'errors' and bail out (as that's not what the user asked for, and it changes the result a lot...)

Reply:
In my tests. If the cloud has an original spacing of >0.01  and you run a feature calculation with the same kernel size {0.01} then it will produce an all NAN sf. i don't really feel like we should stop processing because one of the cloud did not have enough points to calculate a proper sf. That is why i kept a {fall_back_value}.

Or should the user make sure that every cloud has the same valid fields? with select_entities or something like that?

Reply:
Ah ok. Can you detect if they provide a fallback value or not then? So as to be sure they are aware and made it on purpose?

Reply:
well the command will throw an error since the fallback value is a must. Not optional, only the USE_ACTIVE_SF and the rest is optional.

-SS SPATIAL {fall_back_step} USE_ACTIVE_SF {value_at_SF_min} {value_at_SF_max} to use spatial step interpolation on active scalar field